Doc/iss 183/create doc to GitHub webhook (#198)
TeamNovaSoft/discord-bot: docs/github-webhook.md

   ![developer settings](./configuration/github_pat/developer_settings.png)

3. In the left sidebar, under PAT tab, click Tokens (classic). Then click `Generate new token` (classic).
   ![classic token](./configuration/github_pat/classic_token.png)

4. In the "Note" field, give your token a descriptive name.

5. To give your token an expiration, select Expiration, then choose a default option or click Custom to enter a date.

6. Select the scopes you'd like to grant this token. To use your token to access repositories from the command line, select repo. A token with no assigned scopes can only access public information. For more information, see Scopes for OAuth apps.

![pat_classic_form](./configuration/github_pat/pat_classic_form.png)

7. Click Generate token. So the token is available for your app now!

### 4. Finalize Discord Bot Configuration

Once the different configurations have been made to connect discord with the github webhook, some of the data obtained is taken to finish with the configuration and integration of the bot.

#### Steps:

1. Configure the PAT obtained from the github profile in the environment variable `GITHUB_ACCESS_TOKEN`

> 2. The channel that will receive the open PR alert messages is configured by means of the environment variable `GITHUB_PR_REVIEW_CHANNEL`

Reviewer: GITHUB_PR_PUBLISH_ENABLED is another .env that should be updated to enable the feature, should we mention it here?
![2025-03-31-09:00:29-screenshot](https://github.com/user-attachments/assets/12a8e4a1-7708-448b-8f8e-e7f6b21c8acc)

This documentation details the process of setting up an integration between Discord and GitHub, enabling seamless communication and automation of key notifications directly within your Discord channels. By connecting your GitHub repository with Discord, you'll be able to receive instant alerts about opened `Pull requests`. This integration facilitates real-time collaboration and keeps your team informed about repository changes, improving efficiency and responsiveness to project updates. The integration needs some configs like generate `discord webhook`, setup a `github webhook` in a repository and create `Personal Access Token` in github profile (PAT)

The following keys, filled it with the information of your configuration:

```
  GITHUB_PR_REVIEW_CHANNEL = DISCORD_CHANNEL_REVIEW
  GITHUB_ACCESS_TOKEN = YOUR_GITHUB_ACCESS_TOKEN
```

- **GITHUB_PR_REVIEW_CHANNEL**: Discord channel where the bot will send PR review messages.
- **GITHUB_ACCESS_TOKEN**: Your access token for authentication.

> [!NOTE]
> if you want detailed information about this integration process you could check this [doc](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks)

> ## Steps to Enable Integration Github Webhook

Reviewer: We could add a TOC at line 17, what do you think? (table of context or index)

Author: You are right thanks for the advice

Reviewer: Thanks for check!

### 1. Generate discord webhook

> To enable discord webhook, you need to one integration a service account and obtain webhook url where github will send messages about PRs.

Reviewer: I think the message is not clear as it is currently, what do you think about something like this?

```suggestion
To enable Discord's Webhook, you need to go to the integrations section in your server settings and obtain Webhook URL where github will send messages about PRs.
```

Or let me know in spanish what was intended to say with this paragraph maybe we can improve the translation

Thanks in advance

1. In the upper-right corner of any page on GitHub, click your profile photo, then click Settings.
   ![profile settings](./configuration/github_pat/profile_settings.png)

2. In the left sidebar, click Developer settings.
   ![developer settings](./configuration/github_pat/developer_settings.png)

3. In the left sidebar, under PAT tab, click Tokens (classic). Then click `Generate new token` (classic).
   ![classic token](./configuration/github_pat/classic_token.png)

4. In the "Note" field, give your token a descriptive name.

5. To give your token an expiration, select Expiration, then choose a default option or click Custom to enter a date.

6. Select the scopes you'd like to grant this token. To use your token to access repositories from the command line, select repo. A token with no assigned scopes can only access public information. For more information, see Scopes for OAuth apps.

![pat_classic_form](./configuration/github_pat/pat_classic_form.png)

> 7. Click Generate token. So the token is available for your app now!

Reviewer: I think steps above was like collecting information, but was missed a step about what to do with that information, where do we put it on the .envs to make the integration work?

what do you think?
